tests/tests.go

```diff
@@ -18,7 +18,128 @@
+}
+
+func deleteNamespace(namespace string) error {
+	return clientset.CoreV1().Namespaces().Delete(context.TODO(), namespace, metav1.DeleteOptions{})
+}
+
+func removeTestNamespaces() {
+	By(fmt.Sprintf("Waiting for namespace %s to be removed, this can take a while ...\n", testNamespace))
+	EventuallyWithOffset(1, func() bool { return apierrors.IsNotFound(deleteNamespace(testNamespace)) }, 120*time.Second, 5*time.Second).
+		Should(BeTrue(), "Namespace %s haven't been deleted within the given timeout", testNamespace)
+}
+
+func createPrivilegedPodWithIp(podName, nadName, cidr string) {
+	By(fmt.Sprintf("Creating pod %s with priviliged premission and ip %s", podName, cidr))
+	privileged := new(bool)
+	*privileged = true
+	podObject := &corev1.Pod{ObjectMeta: metav1.ObjectMeta{
+		Name: podName,
+		Namespace: testNamespace,
+		Annotations: map[string]string{"k8s.v1.cni.cncf.io/networks": nadName},
+	},
+	Spec: corev1.PodSpec{ Containers: []corev1.Container{{Name: "test",
+		Image:   "alpine",
+		Command: []string{"sh", "-c", fmt.Sprintf("ip address add %s dev net1; sleep 99999", cidr)},
+		SecurityContext: &corev1.SecurityContext{ Privileged: privileged}}}}}
+
+	_, err := clientset.CoreV1().Pods(testNamespace).Create(context.TODO(), podObject, metav1.CreateOptions{})
+	Expect(err).ToNot(HaveOccurred(), fmt.Sprintf("should succeed creating pod %s", podName))
+
+	By("waiting for pod container to be in Running state")
+	Eventually(func() bool {
+		pod, err := clientset.CoreV1().Pods(testNamespace).Get(context.TODO(), podName, metav1.GetOptions{})
+		Expect(err).ToNot(HaveOccurred(), fmt.Sprintf("should succeed getting pod %s instance", podName))
+
+		if len(pod.Status.ContainerStatuses) > 0 {
+			for _, containerStatus := range pod.Status.ContainerStatuses {
+				if containerStatus.Ready != true {
+					return false
+				}
+			}
+
+			return true
+		}
+		return false
+
+	}, 120*time.Second, 5*time.Second).Should(BeTrue(), "should succeed getting pod container to Ready state")
+}
+
+func deletePodsInTestNamespace() {
+	By(fmt.Sprintf("cleaning Pods in %s namespace", testNamespace))
+	podList, err := clientset.CoreV1().Pods(testNamespace).List(context.TODO(), metav1.ListOptions{})
+	Expect(err).ToNot(HaveOccurred(), fmt.Sprintf("should succeed getting pod list in namespace %s", testNamespace))
+
+	for _, pod := range podList.Items {
+		err = clientset.CoreV1().Pods(testNamespace).Delete(context.TODO(), pod.Name, metav1.DeleteOptions{})
+		Expect(err).ToNot(HaveOccurred(), fmt.Sprintf("should succeed delete pod %s", pod.Name))
+	}
+}
+
+func createNetworkAttachmentDefinition(nadName, bridgeName, config string) {
+	By(fmt.Sprintf("Adding NetworkAttachmentDefinition %s of ovs bridge", nadName))
+	nad := &netv1.NetworkAttachmentDefinition{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:        nadName,
+			Namespace:   testNamespace,
+			Annotations: map[string]string{"k8s.v1.cni.cncf.io/resourceName": "ovs-cni.network.kubevirt.io/" + bridgeName },
```

Comment:
Is it mandatory to add this annotation?

Reply:
the purpose of the this annotation is to make sure the pod is assigned to a node that has this ovs bridge. 
Since we have only 1 node, it is not really necessary. do you think I should remove it?

Reply:
Ok I understand now, it is good to have it maybe it worth adding comment about it

Reply:
Done